File: src/net/merayen/elastic/system/actions/LoadProject.java
package net.merayen.elastic.system.actions;

import java.io.File;

import net.merayen.elastic.system.Action;
import net.merayen.elastic.system.intercom.backend.InitBackendMessage;
import net.merayen.elastic.system.intercom.ui.InitUIMessage;

/**
 * Loads a main project.
 */
public class LoadProject extends Action {

	private final String path;

	public LoadProject(String path) {
		this.path = path;
	}

	@Override
	protected void run() {
		//if(!new File(path).isDirectory())
		//	throw new RuntimeException("Project not found");

		system.end();

		system.sendMessageToUI(new InitUIMessage());

		system.sendMessageToBackend(new InitBackendMessage(44100, 16, 512, path));
	}
}

File: src/main/kotlin/net/merayen/elastic/system/actions/LoadProject.kt
package net.merayen.elastic.system.actions

import net.merayen.elastic.system.Action
import net.merayen.elastic.system.intercom.backend.InitBackendMessage
import net.merayen.elastic.system.intercom.ui.InitUIMessage

/**
 * Loads a main project.
 */
class LoadProject(private val path: String) : Action() {
	override fun run() {
		system.end()
		system.sendMessageToUI(listOf(InitUIMessage()))
		system.sendMessageToBackend(listOf(InitBackendMessage(44100, 16, 1024, path)))
	}
}
